### PR TITLE
Implement "stay logged" possibility

### PR DESCRIPTION
**Pull request by Odelpasso**
Hello,

I implemented the possibility for an user to quit MiniGalaxy (with Quit button) stay logged or disconnected. For that, the user must to check or uncheck the check button in Preferences. 

In my case, it works at 90% because i'm quite frustrated. If the user close the windows with the X button at the top right, the process still exist + stay logged if he launch MiniGalaxy again.

If the code is correct, i think for the 1st time i fix an issue : 
https://github.com/sharkwouter/minigalaxy/issues/22

### CODE REVIEW

**Comment by sharkwouter** [comment]
Thanks for the pulI request! It looks pretty good, but I do have some remarks and I might be able to help you out with the issue you have.

I think you'll want to remove this line: https://github.com/sharkwouter/minigalaxy/blob/14587c09ae51bcfc86dc47035cb7232426ce4af5/bin/minigalaxy#L61

Then make an exit function in ``window.py`` and use ``self.connect`` in the init to connect a destroy event to that fucntion. Although I'm not 100% sure if that would work.

You should probably also check if the setting is set on startup and show the login screen if not. You can probably just unset the refresh token and username in the init of ``windows.py`` if it is not set.

Could you rename the ``stay_log`` to ``stay_logged_in`` everywhere? That would make it clearer a bit clearer, sorry :/

**Comment by Odelpasso** [comment]
I modified my commit to check if the user set the option to stay logged.
It was easiest for me to do that instead of the destroy event + `self.connect`

But if you have the possibility to code that and show me how it's implemented  (on Discord), it will be great for my knowledge.

File: minigalaxy/window/preferences.py [hunk 1]
```diff
@@ -45,6 +46,13 @@ def __init__(self, parent, config):
         self.__set_language_list()
         self.button_file_chooser.set_filename(config.get("install_dir"))
         self.switch_install.set_active(self.__config.get("keep_installers"))
+        self.button_stay_log.set_active(self.__config.get("stay_log"))
```
Comment (sharkwouter) [inline minigalaxy/window/preferences.py]: Here it still says stay_log

**Comment by Odelpasso** [comment]
I think is good now.

**Comment by Odelpasso** [comment]
Typo Error. It's good now

File: minigalaxy/window/window.py [hunk 2]
```diff
@@ -44,6 +44,11 @@ def __init__(self, name):
         if not os.path.exists(THUMBNAIL_DIR):
             os.makedirs(THUMBNAIL_DIR)
 
+        # Check if the user unset to stay logged in Preference and refresh token
+        if not self.api.config.get("stay_logged_in"):
```
Comment (sharkwouter) [inline minigalaxy/window/window.py]: This part causes a problem for old installations. An old configuration file will return ``None`` here, but it should be ``True``. Try this code:
```
stay_logged_in = self.api.config.get("stay_logged_in")
        if stay_logged_in is not None and not stay_logged_in:
            self.config.unset("username")
            self.config.unset("refresh_token")
        else:
            self.config.set("stay_logged_in", True)
```

**Comment by Odelpasso** [comment]
Change done. 
I replaced the GtkCheckButton to a GtkSwitch

File: data/ui/preferences.ui [hunk 3]
```diff
@@ -135,6 +135,29 @@
                 <property name="top_attach">2</property>
               </packing>
             </child>
+            <child>
+              <object class="GtkLabel">
+                <property name="visible">True</property>
+                <property name="can_focus">False</property>
+                <property name="halign">start</property>
+                <property name="label" translatable="yes">Stay Logged</property>
```
Comment (sharkwouter) [inline data/ui/preferences.ui]: Can you change the text on the label to ``Stay logged in: ``

**Comment by sharkwouter** [comment]
Looks good! Thanks a lot for the work!

**Pull request merged**